graveler/graveler.go

```diff
@@ -866,6 +882,23 @@ func (g *Graveler) Commit(ctx context.Context, repositoryID RepositoryID, branch
 		if err != nil {
 			return "", fmt.Errorf("get branch: %w", err)
 		}
+
+		// fill commit information - use for pre-commit and after adding the commit information used by commit
+		commit := Commit{
+			Committer:    params.Committer,
+			Message:      params.Message,
+			CreationDate: time.Now(),
+			Metadata:     params.Metadata,
+		}
+		if branch.CommitID != "" {
+			commit.Parents = CommitParents{branch.CommitID}
+		}
+
+		err = g.callPreCommitHooks(ctx, repositoryID, branchID, commit)
+		if err != nil {
+			return "", fmt.Errorf("pre-commit hooks: %w", err)
```

Comment:
I think we should have an `ErrAbortedByHook` (bad name, but you get the idea) that will be propagated when a commit or merge operation fail. This will allow the API to return an informative error (possibly with a link to the failing hook for details, etc)

Reply:
True, as EntryCatalog will implement the actual load/parse/evaluate/execute of the actions it will use a specific error code with the execute/run ID to identify the specific call.

Reply:
Used multi error package to format the final error.
It can be used with errors.Is and ErrAbortedByHook or the hook's error if needed.